packages/ketcher-core/src/domain/entities/DrawingEntitiesManager.ts

```diff
@@ -375,6 +375,18 @@ export class DrawingEntitiesManager {
       command.addOperation(operation);
     }
 
+    if (
+      availableAttachmentPointForBondEnd === 'R2' &&
+      !monomer.hasAttachmentPoint('R1') &&
+      bond.firstMonomer.hasAttachmentPoint('R1')
+    ) {
+      // Prevents forming invalid R2-R2 bonds when second monomer does not have R1 point
+      bond.firstMonomer.removePotentialBonds();
+      bond.firstMonomer.setPotentialBond('R1', bond);
+      const operation = new MonomerHoverOperation(bond.firstMonomer, true);
+      command.addOperation(operation);
+    }
+
```

Comment:
Hi! Not bad, it works! But you also should check for bond.firstMonomer.isAttachmentPointUsed('R1'). Currently you will get infinite number of bonds:
<img width="529" alt="image" src="https://github.com/epam/ketcher/assets/60065704/53fcab6b-8544-4985-87b4-391334b10274">

Also i would suggest to refactor this whole function and make every 'if' return command. This way they will not stack one on another. But it can make additional effort, so maybe we can leave this for future tasks.

Reply:
JFYI each 'if' will be replaced to specific monomer class soon in another PR.

Reply:
(hopefully) fixed.